stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/BackgroundDecorator.kt

```diff
@@ -1,57 +1,85 @@
         viewHolder: MessagePlainTextViewHolder,
         data: MessageListItem.MessageItem
     ) {
-        val radius = DEFAULT_CORNER_RADIUS_DP.dpToPxPrecise()
-        viewHolder.binding.messageText.background = if (data.isMine) {
-            BackgroundDrawable(
-                color = ContextCompat.getColor(viewHolder.itemView.context, MESSAGE_CURRENT_USER_BACKGROUND),
-                topLeftCornerPx = radius,
-                topRightCornerPx = radius,
-                bottomRightCornerPx = if (data.positions.contains(MessageListItem.Position.BOTTOM)) 0f else radius,
-                bottomLeftCornerPx = radius
-            )
-        } else {
-            BackgroundStrokeDrawable(
-                color = ContextCompat.getColor(viewHolder.itemView.context, MESSAGE_OTHER_USER_BACKGROUND),
-                strokeColor = ContextCompat.getColor(viewHolder.itemView.context, MESSAGE_OTHER_STROKE_COLOR),
-                strokeWidth = DEFAULT_STROKE_WIDTH_DP.dpToPxPrecise(),
-                topLeftCornerPx = radius,
-                topRightCornerPx = radius,
-                bottomRightCornerPx = radius,
-                bottomLeftCornerPx = if (data.positions.contains(MessageListItem.Position.BOTTOM)) 0f else radius
-            )
+        setDefaultBackgroundDrawable(viewHolder.binding.messageText, data)
+    }
+
+    private fun setDefaultBackgroundDrawable(view: View, data: MessageListItem.MessageItem) {
+        val radius = DEFAULT_CORNER_RADIUS
+        val bottomRightCorner =
+            if (data.isMine && data.positions.contains(MessageListItem.Position.BOTTOM)) 0f else radius
+        val bottomLeftCorner =
+            if (data.isMine.not() && data.positions.contains(MessageListItem.Position.BOTTOM)) 0f else radius
+        val shapeAppearanceModel =
+            ShapeAppearanceModel.builder().setAllCornerSizes(radius).setBottomLeftCornerSize(bottomLeftCorner)
+                .setBottomRightCornerSize(bottomRightCorner).build()
+        view.background = MaterialShapeDrawable(shapeAppearanceModel).apply {
+            if (data.isMine) {
+                paintStyle = Paint.Style.FILL
+                setTint(ContextCompat.getColor(view.context, MESSAGE_CURRENT_USER_BACKGROUND))
+            } else {
+                paintStyle = Paint.Style.FILL_AND_STROKE
+                setStrokeTint(ContextCompat.getColor(view.context, MESSAGE_OTHER_STROKE_COLOR))
+                strokeWidth = DEFAULT_STROKE_WIDTH
+                setTint(ContextCompat.getColor(view.context, MESSAGE_OTHER_USER_BACKGROUND))
+            }
         }
     }
 
+    override fun decorateOnlyMediaAttachmentsMessage(
+        viewHolder: OnlyMediaAttachmentsViewHolder,
+        data: MessageListItem.MessageItem
+    ) {
+        val cornerRadius = DEFAULT_CORNER_RADIUS - 2.dpToPxPrecise()
+        val bottomRightCornerRadius =
+            if (data.isMine && data.positions.contains(MessageListItem.Position.BOTTOM)) 0f else cornerRadius
```

Comment:
And here, the `bottomRight/Left/cornerRadius` values are not considering the `right-to-left` types of alphabets. The corners will look good for the `left-to-right` alphabets only. For the `right-to-left` ones `bottomRightCornerRadius` and `bottomLeftCornerRadius` should be switched around I suppose (to be displayed as mirror reflection). Right?

Reply:
Yes, you're right again. I prefer iterate over different enhancements. We need to make a design as is in Figma firstly.